app/controllers/puppetclass_lookup_keys_controller.rb

```diff
@@ -4,8 +4,11 @@ class PuppetclassLookupKeysController < LookupKeysController
   before_action :setup_search_options, :only => :index
 
   def index
-    @lookup_keys = resource_base_search_and_page(:param_classes).smart_class_parameters
-    @puppetclass_authorizer = Authorizer.new(User.current, :collection => @lookup_keys.map {|key| key.param_class.try(:id)}.compact.uniq)
+    @lookup_keys = resource_base_search_and_page.distinct.smart_class_parameters.preload(:lookup_values)
```

Comment:
Since we are going directly to EnvironmentClass, why do we need `smart_class_parameters` here?